bootstrap/secret/insecure.go

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package secret
+
+import (
+	"errors"
+	"fmt"
+
+	"strings"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/interfaces"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+// InsecureProvider implements the SecretProvider interface for insecure secrets
+type InsecureProvider struct {
+	lc            logger.LoggingClient
+	configuration interfaces.Configuration
+	lastUpdated   time.Time
+}
+
+// NewInsecureProvider creates, initializes Provider for insecure secrets.
+func NewInsecureProvider(config interfaces.Configuration, lc logger.LoggingClient) *InsecureProvider {
+	return &InsecureProvider{
+		configuration: config,
+		lc:            lc,
+		lastUpdated:   time.Now(),
+	}
+}
+
+// GetSecrets retrieves secrets from a Insecure Secrets secret store.
+// path specifies the type or location of the secrets to retrieve.
+// keys specifies the secrets which to retrieve. If no keys are provided then all the keys associated with the
+// specified path will be returned.
+func (p *InsecureProvider) GetSecrets(path string, keys ...string) (map[string]string, error) {
+	results := make(map[string]string)
+	pathExists := false
+	var missingKeys []string
+
+	insecureSecrets := p.configuration.GetInsecureSecrets()
+	if insecureSecrets == nil {
+		err := fmt.Errorf("InsecureSecrets missing from configuration")
```

Comment:
is missing from configuration the only possibility to result in `nil`?  What about empty secret list?

Reply:
Yes.